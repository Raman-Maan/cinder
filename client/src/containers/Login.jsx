import React, { Component } from 'react';
import { Container, Form, FormGroup, Label, Input, Button } from 'reactstrap';
import { Redirect, Link } from 'react-router-dom';

import './styles/Login.css';
import logo from '../assets/logo.svg';

class Login extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
      loggedIn: false,
      loginBtnText: 'Log In'
    };

    this.setInput = this.setInput.bind(this);
    this.userLogin = this.userLogin.bind(this);
  }

  setInput(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  userLogin() {
    // TODO - validate fields

    const payload = { email: this.state.email, password: this.state.password };

    this.setState({ loginBtnText: 'Logging In...' });

    // attempt to log in
    fetch('/api/user/login', {
      method: 'POST',
      body: JSON.stringify(payload)
    })
      .then(res => {
        // all good, let's go to home page
        if (res.status === 200) {
          this.setState({ loginBtnText: 'Log In', loggedIn: true });
        } else {
          throw new Error(`Login Error: ${res.status}`);
        }
      })
      .catch(err => {
        this.setState({ loginBtnText: 'Log In' });
        // TODO - better errors for user - what went wrong?
        switch (err) {
          default:
            console.error(err);
        }
      });
  }

  render() {
    if (this.state.loggedIn) {
      return <Redirect to="/" />;
    }

    return (
      <div className="Login">
        <Container>
          <div className="logo">
            <h1 className="display-3">cinder</h1>
            <img src={logo} className="App-logo" alt="logo" />
          </div>
          <hr />

          <Form className="login-form">
            <FormGroup>
              <Label for="email">Email</Label>
              <Input
                required
                type="email"
                name="email"
                placeholder="Email"
                onChange={this.setInput}
              />
            </FormGroup>

            <FormGroup>
              <Label for="password">Password</Label>
              <Input
                required
                type="password"
                name="password"
                onChange={this.setInput}
                placeholder="Password"
              />
            </FormGroup>

            <FormGroup>
              <Button
                outline
                block
                color="dark"
                type="button"
                onClick={this.userLogin}
              >
                {this.state.loginBtnText}
              </Button>

              <Button outline block color="secondary" tag={Link} to="/signup">
                Sign Up
              </Button>
            </FormGroup>
          </Form>
        </Container>
      </div>
    );
  }
}

export default Login;
