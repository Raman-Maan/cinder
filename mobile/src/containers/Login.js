import React from 'react';
import Expo from 'expo';
import { StyleSheet, Text } from 'react-native';
import { Container, H1, Content, Form, Item, Label, Input, Button } from 'native-base';

export default class Login extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
    };
  }

  userLogin() {
    fetch('/api/user/login', {
      method: 'POST',
      body: JSON.stringify({
        email: this.state.email,
        password: this.state.password,
      }),
    }).then((res) => {
      // handle response here
      if (res.status === 200) {
        // all good
        // execute login here
      } else {
        // handle error here
        throw new Error(res.status);
      }
    }).catch(err => err);
  }

  render() {
    return (
      <Container style={styles.container}>
        <Content style={styles.content}>
          <H1 center>cinder</H1>
          <Form>
            <Item floatingLabel>
              <Label>Email</Label>
              <Input onChangeText={txt => this.setState({ email: txt })} />
            </Item>
            <Item floatingLabel>
              <Label>Password</Label>
              <Input secureTextEntry onChangeText={txt => this.setState({ password: txt })} />
            </Item>
            <Button bordered primary block >
              <Text>Log In</Text>
            </Button>
            <Button bordered dark block>
              <Text>Sign Up</Text>
            </Button>
          </Form>
        </Content>
      </Container>
    );
  }
}


const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  content: {
    paddingLeft: 10,
    paddingRight: 10,
  },
});
