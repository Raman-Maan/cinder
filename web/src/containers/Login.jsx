import React, { Component } from 'react';
import { Container, Form, FormGroup, Label, Input, Button } from 'reactstrap';
import { Redirect, Link } from 'react-router-dom';

import './styles/Login.css';
import logo from '../assets/logo.svg';

class Login extends Component {
  constructor(props) {
    super(props);

    this.setInput = this.setInput.bind(this);
    this.userLogin = this.userLogin.bind(this);

    this.state = {
      email: '',
      password: '',
      loggedIn: false,
      loginBtnText: 'Log In'
    };
  }

  setInput(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  userLogin(e) {
    e.preventDefault();

    this.setState({ loginBtnText: 'Logging In...' });
    fetch('/api/login', {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-type': 'application/json'
      },
      body: JSON.stringify({
        email: this.state.email,
        password: this.state.password
      })
    })
      .then(res => res.json())
      .then(res => {
        if (res.status === 200) {
          this.setState({ loginBtnText: 'Log In', loggedIn: true });
          localStorage.setItem('token', res.token);
        } else {
          throw new Error(`${res.status} ${res.err}`);
        }
      })
      .catch(err => {
        this.setState({ loginBtnText: 'Log In' });
        // TODO - better errors for user - what went wrong?
        alert(err);
      });
  }

  render() {
    if (this.state.loggedIn) {
      return <Redirect to="/" />;
    }

    return (
      <div className="Login">
        <Container>
          <div className="logo">
            <h1 className="display-3">cinder</h1>
            <img src={logo} className="App-logo" alt="logo" />
          </div>
          <hr />

          <Form className="login-form" onSubmit={e => this.userLogin(e)}>
            <FormGroup>
              <Label for="email">Email</Label>
              <Input
                required
                id="email"
                type="email"
                name="email"
                placeholder="Email"
                onChange={this.setInput}
              />
            </FormGroup>

            <FormGroup>
              <Label for="password">Password</Label>
              <Input
                required
                id="password"
                type="password"
                name="password"
                onChange={this.setInput}
                placeholder="Password"
              />
            </FormGroup>

            <FormGroup>
              <Button id="submitBtn" outline block color="dark" type="submit">
                {this.state.loginBtnText}
              </Button>

              <Button
                id="signUpBtn"
                outline
                block
                color="secondary"
                tag={Link}
                to="/signup"
              >
                Sign Up
              </Button>
            </FormGroup>
          </Form>
        </Container>
      </div>
    );
  }
}

export default Login;
