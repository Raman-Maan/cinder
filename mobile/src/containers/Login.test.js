import React from 'react';
import renderer from 'react-test-renderer';

import Login from './Login';

it('renders without crashing', () => {
  const rendered = renderer.create(<Login />).toJSON();
  expect(rendered).toMatchSnapshot();
});
