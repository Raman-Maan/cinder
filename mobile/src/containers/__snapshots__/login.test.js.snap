// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
  virtual={undefined}
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={undefined}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        },
        Object {
          "paddingLeft": 10,
          "paddingRight": 10,
        },
      ]
    }
    viewIsInsideTabBar={false}
    virtual={undefined}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        center={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontSize": 27,
            "lineHeight": 32,
          }
        }
        virtual={undefined}
      >
        cinder
      </Text>
      <View
        style={Object {}}
        virtual={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 1,
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "flexDirection": "row",
              "marginLeft": 15,
              "marginTop": 15,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "left": 0,
                "opacity": 1,
                "paddingBottom": undefined,
                "paddingTop": undefined,
                "position": "absolute",
                "right": 0,
                "top": 18,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#575757",
                  "fontSize": 17,
                  "left": 0,
                  "paddingRight": 5,
                  "top": 6,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 17,
                  }
                }
                virtual={undefined}
              >
                Email
              </Text>
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholderTextColor="#575757"
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 17,
                "height": 50,
                "lineHeight": 24,
                "marginLeft": null,
                "paddingLeft": null,
                "paddingRight": 5,
                "top": 10,
              }
            }
            underlineColorAndroid="rgba(0,0,0,0)"
            value=""
            virtual={undefined}
          />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 1,
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "flexDirection": "row",
              "marginLeft": 15,
              "marginTop": 15,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "left": 0,
                "opacity": 1,
                "paddingBottom": undefined,
                "paddingTop": undefined,
                "position": "absolute",
                "right": 0,
                "top": 18,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#575757",
                  "fontSize": 17,
                  "left": 0,
                  "paddingRight": 5,
                  "top": 6,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 17,
                  }
                }
                virtual={undefined}
              >
                Password
              </Text>
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholderTextColor="#575757"
            secureTextEntry={true}
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 17,
                "height": 50,
                "lineHeight": 24,
                "marginLeft": null,
                "paddingLeft": null,
                "paddingRight": 5,
                "top": 10,
              }
            }
            underlineColorAndroid="rgba(0,0,0,0)"
            value=""
            virtual={undefined}
          />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "transparent",
              "borderBottomWidth": 1,
              "borderColor": "#007aff",
              "borderLeftWidth": 1,
              "borderRadius": 5,
              "borderRightWidth": 1,
              "borderTopWidth": 1,
              "borderWidth": 1,
              "elevation": null,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Log In
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "transparent",
              "borderBottomWidth": 1,
              "borderColor": "#000",
              "borderLeftWidth": 1,
              "borderRadius": 5,
              "borderRightWidth": 1,
              "borderTopWidth": 1,
              "borderWidth": 1,
              "elevation": null,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Sign Up
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
